File: javascript/for-loop.js
// npm install prompt-sync
const prompt = require('prompt-sync')();

const solution = Math.floor(Math.random() * 100) + 1;

console.log("You have 10 tries to guess my number between 1 and 100.\n");
for (let i = 1; i <= 10; i++) {
	let guess = prompt(`[${i}/10] What's your guess? `);
	if (guess < solution) {
		console.log("My number is bigger.");
	} else if (guess > solution) {
		console.log("My number is smaller.");
	} else {
		console.log("That's it. You guessed it.");
		process.exit(0);
	}
}

process.exit(1)

File: cpp/for-loop.cpp
#include <iostream>
#include <string>
#include <time.h>
#include <stdlib.h>

using namespace std;

int main() {
	srand (time(NULL));
	const int solution = rand() % 100 + 1;
	cout << "You have 10 tries to guess my number between 1 and 100." << endl;

	for (int i = 1; i <= 10; i++) {
		cout << "[" << i << "/10] What's your guess?" << endl;		

		string input;
		getline(cin, input);
		int guess = stoi(input);

		if (guess < solution) {
			cout << "My number is bigger." << endl;
		} else if (guess > solution) {
			cout << "My number is smaller." << endl;
		} else {
			cout << "That's it. You guessed it." << endl;
			return 0;
		}
	}
	
	cout << "Ohh you didn't guess it. :(" << endl;
	return 1;
}